### PR TITLE
Add get file method

### PR DESCRIPTION
**Pull request by alyosha**
# Summary
Decided to make a new PR since #292 was stale, needed a rebase, and because I made a few different implementation choices. Would love to see this method available for a number of personal projects I'm working on. 

Went ahead and addressed your feedback from the previous PR @james-lawrence and added a test, addressed the segfault (plus one additional case), and also split out the status code error handling into its own helper method. 

I am not a hundred percent confident with the test, so would appreciate any feedback/thought you might have.

Happy to make any changes required 🙇

### CODE REVIEW

**Comment by alyosha** [comment]
@james-lawrence any chance at getting a review? would love to use this in an upcoming proj

File: files.go [hunk 1]
```diff
@@ -187,6 +187,11 @@ func (api *Client) GetFileInfoContext(ctx context.Context, fileID string, count,
 	return &response.File, response.Comments, &response.Paging, nil
 }
 
+// GetFile retreives a given file from its private download URL
+func (api *Client) GetFile(file *File, writer io.Writer) error {
```
Comment (james-lawrence) [inline files.go]: instead of passing in the file object, just pass in the string url.

**Review by james-lawrence — CHANGES_REQUESTED** [review]
minor change for the api.

as for the test.... i mean it doesn't really test anything except that initial if statement. and that the request received a response. but its fine.

**Comment by alyosha** [comment]
yeah, I didn't think there was much else to test since it's mostly dependent on the response from the http req. happy to add any cases you can think but in the meantime made the API fix you suggested 🙏 thanks for the quick response

**Review by james-lawrence — APPROVED** [review]
LGTM.

**Comment by james-lawrence** [comment]
just need to fix your test =)

**Comment by alyosha** [comment]
haha I realized it right as I stepped in the shower... thanks so much for the review!

**Pull request merged**